met-web/tests/unit/components/feedbackListing.test.tsx

```diff
@@ -0,0 +1,146 @@
+    ...jest.requireActual('components/common'),
+    PrimaryButton: ({ children, ...rest }: { children: ReactNode; [prop: string]: unknown }) => {
+        return <button {...rest}>{children}</button>;
+    },
+}));
+
+describe('Feedback Listing tests', () => {
+    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => jest.fn());
+    jest.spyOn(notificationSlice, 'openNotification').mockImplementation(jest.fn());
+    const getFeedbackPageMock = jest.spyOn(feedbackService, 'getFeedbacksPage');
+
+    beforeEach(() => {
+        setupEnv();
+    });
+
+    test('Feedback table is rendered and feedback is fetched', async () => {
+        getFeedbackPageMock.mockReturnValue(
+            Promise.resolve({
+                items: [mockFeedbackOne, mockFeedbackTwo],
+                total: 2,
+            }),
+        );
+        render(<FeedbackListing />);
+
+        await waitFor(() => {
+            expect(screen.getByText('Feedback One')).toBeInTheDocument();
+            expect(screen.getByText('2022-09-17')).toBeInTheDocument();
+
+            expect(screen.getByText('Feedback Two')).toBeInTheDocument();
+            expect(screen.getByText('2022-09-19')).toBeInTheDocument();
+        });
+    });
+
+    test('Feedback table is empty', async () => {
+        render(<FeedbackListing />);
+        const feedbackListing = screen.getByTestId('listing-table');
+
+        getFeedbackPageMock.mockReturnValue(
+            Promise.resolve({
+                items: [],
+                total: 0,
+            }),
+        );
+        render(<FeedbackListing />);
+
+        await waitFor(() => {
+            expect(feedbackListing).toBeVisible();
+        });
+    });
+
+    test('fetchs feedback with the sort key as the rating', async () => {
+        getFeedbackPageMock.mockReturnValue(
+            Promise.resolve({
+                items: [mockFeedbackOne],
+                total: 1,
+            }),
+        );
+        render(<FeedbackListing />);
+
+        await waitFor(() => {
+            expect(screen.getByText('Feedback One')).toBeInTheDocument();
+        });
+
+        await waitFor(() => {
+            expect(getFeedbackPageMock).lastCalledWith({
+                page: 1,
+                size: 10,
+                sort_key: 'rating',
+                sort_order: 'asc',
+            });
+        });
+    });
+
+    test('Test feedback table pagination', async () => {
+        getFeedbackPageMock.mockReturnValue(
+            Promise.resolve({
+                items: [
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                    mockFeedbackOne,
+                ],
```

Comment:
can u do a for loop or  Array.from({length:11} syntax or array.fill to do the same thing rather than writing this 11 times? i think there will be  lot of different ways to achieve this..

Reply:
Thank you I never thought about that. Cleaned that up!